src/main/java/com/okta/sdk/framework/FilterBuilder.java

```diff
@@ -204,24 +168,72 @@ public FilterBuilder lessThanOrEqual(DateTime value) {
     }
 
     public FilterBuilder and() {
-        STRING_BUILDER.append(AND_SIGN);
+        filterBuilder.append(AND_SIGN);
+        totalNumberOfLogicalOperators += 1;
+        checkComplexity();
         return this;
     }
 
     public FilterBuilder and(FilterBuilder filterBuilder) {
-        STRING_BUILDER.append(AND_SIGN);
+        this.filterBuilder.append(AND_SIGN);
         return where(filterBuilder);
     }
 
     public FilterBuilder or() {
-        orCount++;
-        STRING_BUILDER.append(OR_SIGN);
+        filterBuilder.append(OR_SIGN);
+        totalNumberOfLogicalOperators += 1;
+        checkComplexity();
         return this;
     }
 
     public FilterBuilder or(FilterBuilder filterBuilder) {
-        orCount++;
-        STRING_BUILDER.append(OR_SIGN);
+        this.filterBuilder.append(OR_SIGN);
         return where(filterBuilder);
     }
+
+    private FilterBuilder equalTo() {
+        filterBuilder.append(EQUAL_SIGN);
+        return this;
+    }
+
+    private FilterBuilder contains() {
+        filterBuilder.append(CONTAIN_SIGN);
+        return this;
+    }
+
+    private FilterBuilder startsWith() {
+        filterBuilder.append(STARTS_WITH_SIGN);
+        return this;
+    }
+
+    private FilterBuilder greaterThan() {
+        filterBuilder.append(GREATER_THAN_SIGN);
+        return this;
+    }
+
+    private FilterBuilder greaterThanOrEqual() {
+        filterBuilder.append(GREATER_THAN_OR_EQUAL_SIGN);
+        return this;
+    }
+
+    private FilterBuilder lessThan() {
+        filterBuilder.append(LESS_THAN_SIGN);
+        return this;
+    }
+
+    private boolean checkComplexity() {
```

Comment:
I think it would be fine to doc the current restrictions on the developer site (if we don't already), but enforcing them in the client code is going too far IMO. I think we should remove it and let the server drive the business logic.